front/components/data_source/DataSourceEdition.tsx

```diff
@@ -0,0 +1,436 @@
+  ) => void;
+  setIsLoadingByProvider?: (
+    providers: (
+      prev: Record<ConnectorProvider, boolean | undefined>
+    ) => Record<ConnectorProvider, boolean | undefined>
+  ) => void;
+}
+
+const REDIRECT_TO_EDIT_PERMISSIONS = [
+  "confluence",
+  "google_drive",
+  "microsoft",
+  "slack",
+  "intercom",
+];
+
+function isDataSourceIntegration(
+  integration: DataSourceIntegration | DataSourceType | null
+): integration is DataSourceIntegration {
+  return (
+    integration !== null &&
+    "id" in integration &&
+    "name" in integration &&
+    "type" in integration &&
+    "config" in integration
+  );
+}
+
+export function DataSourceEditionModal({
+  isOpen,
+  owner,
+  connectorProvider,
+  dataSourceIntegration,
+  onClose,
+  router,
+  dustClientFacingUrl,
+  user,
+  setIsRequestDataSourceModalOpen,
+  setDataSourceIntegrations,
+  setIsLoadingByProvider,
+}: DataSourceEditionModalProps) {
+  const sendNotification = useContext(SendNotificationsContext);
+  const [showConfirmDialog, setShowConfirmDialog] = useState(false);
+
+  if (!connectorProvider || !dataSourceIntegration) {
+    return;
+  }
+
+  const connectorConfiguration = CONNECTOR_CONFIGURATIONS[connectorProvider];
+
+  let isSetup: boolean | null = null;
+  let dataSourceName: string | null = null;
+  if (isDataSourceIntegration(dataSourceIntegration)) {
+    isSetup = !!dataSourceIntegration?.connector;
+    dataSourceName = dataSourceIntegration.dataSourceName;
+  } else {
+    isSetup = true;
+    dataSourceName = dataSourceIntegration.name;
+  }
+
+  let dataSourceOwner: EditedByUser | null | undefined = null;
+  let isDataSourceOwner: boolean = false;
+  if (isSetup) {
+    dataSourceOwner = dataSourceIntegration.editedByUser;
+    isDataSourceOwner =
+      dataSourceIntegration?.editedByUser?.userId === user.sId;
+  }
+
+  const handleEnableManagedDataSource = async (
+    dataSourceIntegration: DataSourceIntegration
+  ) => {
+    if (!setIsLoadingByProvider || !setDataSourceIntegrations) {
+      return;
+    }
+    try {
+      const provider = dataSourceIntegration.connectorProvider;
+      const suffix = dataSourceIntegration.setupWithSuffix;
+      const connectionIdRes = await setupConnection({
+        dustClientFacingUrl,
+        owner,
+        provider,
+      });
+      if (connectionIdRes.isErr()) {
+        throw connectionIdRes.error;
+      }
+      onClose();
+      setIsLoadingByProvider((prev) => ({ ...prev, [provider]: true }));
+
+      const res = await fetch(
+        suffix
+          ? `/api/w/${
+              owner.sId
+            }/data_sources/managed?suffix=${encodeURIComponent(suffix)}`
+          : `/api/w/${owner.sId}/data_sources/managed`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            provider,
+            connectionId: connectionIdRes.value,
+            name: undefined,
```

Comment:
It feels weird to pass undefined values here

Reply:
Only providing `connectionId` should be enough IIRC.

Reply:
Looking into `PostManagedDataSourceRequestBodySchema` you actually need to pass `configuration` and `name`